File: C-Abi-Bright-Aot/Tests/C-Abi-Bridge.Aot.Vb.Tests/TestCrypto/TestCryptoAes/CryptoAesTest.vb
Option Strict On
Option Explicit On

Namespace michele.natale.Tests
  Partial Friend NotInheritable Class CryptoAesTest
    Public Shared Sub StartNative(rounds As Int32)
      TestAesFile(rounds)
      TestAesBytes(rounds)
      TestAesBytesStress()
      Console.WriteLine()
    End Sub
  End Class
End Namespace

File: C-Abi-Bridge-Aot/Tests/CAbiBridge.Aot.Cpp.Tests/crypto_aes_test.cpp

#include "pch.h"
#include <cstdlib>

#include "crypto_aes_test.h"



namespace michele::natale::Tests
{
  static void start_native_aes_crypto(int rounds)
  {
    test_aes_file(rounds);
    test_aes_bytes(rounds);
    test_aes_bytes_stress();

    std::cout << "\n";
  }

  void start_aes_native(int rounds)
  {
    start_native_aes_crypto(rounds);
  }
}